README.org

```diff
@@ -133,11 +143,11 @@ Synchronously download a JPEG file, then create an Emacs image object from the d
    ~FINALLY~ is an optional function called without argument after ~THEN~ or ~ELSE~, as appropriate.  For synchronous requests, this argument is ignored.
 
    ~CONNECT-TIMEOUT~ and ~TIMEOUT~ are a number of seconds that limit how long it takes to connect to a host and to receive a response from a host, respectively.
-      
-   If ~STREAM~ is non-nil, a ~THEN~ callback must be given and the response will be streamed in chunks.  The ~THEN~ function is invoked for each incoming chunk, receiving the current chunk as its parameter.  The type of object passed to the THEN callback depends on the AS argument.  When using this option, the buffering in the underlying curl output stream is turned off, and there are no guarantees regarding the chunk, such as being line-based or not.  It is the user's responsibility to understand and process each chunk, and to construct the finalized response if necessary.  Use the ~FINALLY~ callback to determine when the request is complete.
 
    ~NOQUERY~ is passed to ~make-process~, which see.
 
+   When the HTTP response is streamed, the buffering in the curl output stream is turned off and the PROCESS-FILTER may be called multiple times, depending on the size of the HTTP body.  It is the user's responsibility to understand and process each chunk, and to construct the finalized response if necessary.  There are no guarantees regarding the chunk, such as being line-based or not.
```

Comment:
I got this from the curl manual and tried to summarize it, not doing a good job. This is what the manual says:
```
-N/--no-buffer
    Disables the buffering of the output stream. In normal work situations, curl will use a standard buffered output stream that will have the effect that it will output the data in chunks, not necessarily exactly when the data arrives. Using this option will disable that buffering.

    Note that this is the negated option name documented. You can thus use --buffer to enforce the buffering. 
```
Maybe we should not mention it here, or if we mention it, it could also become an additional option to :as stream.

In practice `--no-buffer` returns data as it arrives (which is what you want probably when chatting with an LLM), otherwise the response is buffered (on my machine it looked like a 4K buffer was used). With `--no-buffer` the user provided PROCESS-FILTER fn is called more often.